pkg/fornaxcore/grpc/server/server.go

```diff
@@ -0,0 +1,153 @@
+package server
+
+import (
+	"centaurusinfra.io/fornax-serverless/pkg/fornaxcore/grpc"
+	"context"
+	"fmt"
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc/peer"
+	"net"
+	"sync"
+)
+
+const FornaxCoreChanSize = 2
+
+type FornaxCoreNodeServer interface {
+	grpc.FornaxCoreServiceServer
+	MessageDispatcher
+}
+
+type grpcServer struct {
+	sync.RWMutex
+	chans map[string]chan<- *grpc.FornaxCoreMessage
+
+	nodeMonitor       NodeMonitor
+	podMonitor        PodMonitor
+	appSessionMonitor AppSessionMonitor
+	grpc.UnimplementedFornaxCoreServiceServer
+}
+
+func (g *grpcServer) enlistNode(node string, ch chan<- *grpc.FornaxCoreMessage) {
+	g.Lock()
+	defer g.Unlock()
+	g.chans[node] = ch
+}
+
+func (g *grpcServer) delistNode(node string) {
+	g.Lock()
+	defer g.Unlock()
+	close(g.chans[node])
+	delete(g.chans, node)
+}
+
+func (g *grpcServer) GetMessage(empty *empty.Empty, server grpc.FornaxCoreService_GetMessageServer) error {
+	peer, ok := peer.FromContext(server.Context())
+	if !ok {
+		return fmt.Errorf("failed to identify peer client")
+	}
+	node, err := net.ResolveTCPAddr(peer.Addr.Network(), peer.Addr.String())
+	if err != nil {
+		// todo: log error
+		return err
+	}
+	nodeIP := node.IP.String()
+
+	chDone := server.Context().Done()
+	ch := make(chan *grpc.FornaxCoreMessage, FornaxCoreChanSize)
```

Comment:
can we check if this node already have a channel created, and reuse it, otherwise it will make a new channel every time and old channels are accumulated every time get message is called and not used anymore.

Reply:
thanks for raising this issue; mere overriding the slot with new channel did have problem.
will modify code to prevent multiple streams between one node to fornax core.